libnemo-private/nemo-file.c

```diff
@@ -4077,6 +4128,12 @@ nemo_file_get_display_name (NemoFile *file)
 	return g_strdup (nemo_file_peek_display_name (file));
 }
 
+char *
+nemo_file_get_extension_name (NemoFile *file)
+{
+	return nemo_file_peek_extension_name (file);
+}
```

Comment:
Here you're returning a new string (`char *` return), so you would need to g_strdup the peeked extension name.